crates/blockifier/src/execution/contract_class.rs

```diff
@@ -546,13 +548,28 @@ impl SierraContractClassV1 {
         Some(contract_entrypoint_to_entrypoint_selector(entrypoint))
     }
 
-    pub fn try_from_json_string(
-        raw_contract_class: &str,
-    ) -> Result<SierraContractClassV1, ProgramError> {
+    pub fn try_from_json_string(raw_contract_class: &str) -> Result<Self, ProgramError> {
+        fn compile(sierra_program: &SierraProgram) -> AotNativeExecutor {
+            let native_context = cairo_native::context::NativeContext::new();
+            let native_program = native_context.compile(sierra_program, None).unwrap();
+            AotNativeExecutor::from_native_module(native_program, OptLevel::Default)
+        }
+
         let sierra_contract_class: SierraContractClass = serde_json::from_str(raw_contract_class)?;
-        let contract_class: SierraContractClassV1 = sierra_contract_class.try_into()?;
+        // todo(rodro): we are having two instances of a sierra program, one it's object form
+        // and another in its felt encoded form. This can be avoided by either:
+        //   1. Having access to the encoding/decoding functions
+        //   2. Refactoring the code on the Cairo mono-repo
 
-        Ok(contract_class)
+        let sierra_program = sierra_contract_class.extract_sierra_program().unwrap();
+        let executor = compile(&sierra_program);
```

Comment:
This looks like the first compilation is happening in the blockifier. Can we move it outside the Blockifier (i.e. Juno)

Reply:
The main purpose of this PR is to give the node control over caching of compiled contracts, whether compiled to native or to the VM. Where the compilation resides is secondary. What is important is that the node decides when the compilation takes place.  As for why the native compilation is done in `try_from_json_string` is because that is also how it's done for the compilation of V0 and V1 contracts to the VM.

While technically it is possible to take out `try_from_json_string` and move it to the node, I think that in its current state it makes more sense for compilation to be in the blockifier. The way the contract class describes executors is very much tied to how a contract is compiled and how that compilation is loaded into memory so splitting it up now is not going to help. 

To be able to move compilation further out into the node will require support from the blockifier and is something for a later PR.